src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,69 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final String DEFAULT_EXCEPTION_TEXT = "Can't register this user. ";
+    private static final String EMPTY_STRING = "";
+    private static final int PASSWORD_MIN_LENGTH = 6;
+    private static final int MIN_USER_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        validateUser(user);
+        validateUserNonDuplicates(user);
+        validateUserFields(user);
+        return storageDao.add(user);
+    }
+
+    private void validateUser(User user) {
+        if (user == null) {
+            throwException("User can't be null");
+        }
+    }
+
+    private void validateUserFields(User user) {
+        validateLogin(user);
+        validatePassword(user);
+        validateUserAge(user);
+    }
+
+    private void validateLogin(User user) {
+        if (user.getLogin() == null) {
+            throwException("User login can't be null");
+        }
+        if (user.getLogin().equals(EMPTY_STRING)) {
+            throwException("Login can't be empty");
+        }
+    }
+
+    private void validatePassword(User user) {
+        if (user.getPassword() == null) {
+            throwException("User password can't be null");
+        }
+        if (user.getPassword().equals(EMPTY_STRING)) {
+            throwException("Password can't be empty");
+        }
```

Comment:
I guess it’s  redundant  conditional because  you check it in next conditional block